src/reporters/critical.ts

```diff
@@ -59,9 +59,35 @@ export const createCriticalStyleStream = (def: StyleDefinition) => {
 
       const chunkData = Buffer.from(process(chunk.toString('utf-8'), line, styleCallback), 'utf-8');
 
+      const chunkDataStr = chunkData.toString('utf-8');
+
+      const lastOpeningBrace = chunkDataStr.lastIndexOf('<');
+
+      if (lastOpeningBrace !== -1 && chunkDataStr.substr(lastOpeningBrace, 6) === '<style') {
```

Comment:
I am not sure this condition is important. From what I can understand only the second one provides value - ie not allowing injecting style into style.

Look like the problem occurs only if chunk contains a piece of _not closed_ style tag __and__ some HTML as that would lead to "style in style" and break HTML

This raises another moment - what else will render into the same problem. And that would be <script> tag.
Unlike style, script can contain some HTML, and if it's extracted and injected in the beginning on the chunk - script will be broken.

There are two possible solutions for this:
- improve in small steps: make it work for styles first, worry about scripts later
- change the script a little bit to track open/closed style/script outside of callback, so the script block spanning multiple chunks will be handled.

Let's go the first way 😉 , and look like your solution is what needed

Reply:
I think you are right about the first condition, have removed it.